File: Torneos.API/Entities/Tournament.cs
using Torneos.API.Entities;

namespace Torneos.API.Entities;

// ⚙️ 1. Definimos las opciones de formato para el desplegable del administrador
public enum TournamentFormat
{
    League,             // Liga todos contra todos
    Knockout,           // Eliminatoria directa (Playoffs)
    GroupsAndPlayoffs   // Fase de grupos + Eliminatoria (estilo Champions)
}

// ⚙️ 2. Definimos las opciones de dónde se juegan los partidos
public enum VenueType
{
    HomeAndAway,  // Ida y vuelta en el campo de cada equipo (Fútbol tradicional)
    SingleRound,  // Una sola vuelta en el campo del local
    NeutralVenue  // Todo en sede neutral (Pistas de tenis, polideportivo municipal, etc.)
}

public enum TournamentStatus
{
    RegistrationOpen,
    InProgress,
    Finished
}

public class Tournament
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    
    // 🔄 CAMBIO: Ahora el tipo es un Enum inteligente en vez de un string plano
    public TournamentFormat Format { get; set; } = TournamentFormat.League; 
    
    // 🆕 NUEVO: Configuración de la sede del torneo
    public VenueType VenueConfig { get; set; } = VenueType.HomeAndAway;

    // 🆕 NUEVO: El candado de prestigio mínimo para poder apuntarse
    public int MinPrestigeRequired { get; set; } = 0; // Por defecto 0 (cualquiera entra)

    // 🆕 NUEVO: Mínimo de jugadores por equipo para poder inscribirse
    public int MinPlayersPerTeam { get; set; } = 0; // Por defecto 0 (sin requisito)

    public TournamentStatus Status { get; set; } = TournamentStatus.RegistrationOpen;
    public bool IsFixtureGenerated { get; set; } = false;

    public int SportId { get; set; }
    public Sport? Sport { get; set; }
    
    public List<Match> Matches { get; set; } = new List<Match>();
    
    public List<TeamTournament> TeamTournaments { get; set; } = new List<TeamTournament>();

    // 🆕 NUEVO: Conexión con la tabla intermedia de estadios asignados al torneo (Sedes Neutras)
    public List<TournamentStadium> TournamentStadiums { get; set; } = new List<TournamentStadium>();
}

File: TorneosWeb/src/app/models/tournament.ts
import { Sport } from './sport';
import { Match } from './match';
import { TeamTournament } from './team-tournament';
import { TournamentStadium } from './tournament-stadium';

export enum TournamentFormat {
  League = 0,
  Knockout = 1,
  GroupsAndPlayoffs = 2
}

export enum VenueType {
  HomeAndAway = 0,
  SingleRound = 1,
  NeutralVenue = 2
}

export enum TournamentStatus {
  RegistrationOpen = 0,
  InProgress = 1,
  Finished = 2
}

export interface Tournament {
  id: number;
  name: string;
  format: TournamentFormat;
  venueConfig: VenueType;
  minPrestigeRequired: number;
  minPlayersPerTeam: number;
  status: TournamentStatus;
  isFixtureGenerated: boolean;
  sportId: number;
  sport?: Sport;
  matches?: Match[];
  teamTournaments?: TeamTournament[];
  tournamentStadiums?: TournamentStadium[];
  _enrolledCount?: number;
  _enrolledTeams?: Team[];
}
import { Team } from './team';
